src/main/java/com/faforever/client/fx/WindowController.java

```diff
@@ -60,12 +58,11 @@ public WindowController(UiService uiService) {
   }
 
   public static Rectangle2D getVisualBounds(Stage stage) {
-    double x1 = stage.getX() + (stage.getWidth() / 2);
-    double y1 = stage.getY() + (stage.getHeight() / 2);
-
-    Rectangle2D windowCenter = new Rectangle2D(x1, y1, 1, 1);
-    ObservableList<Screen> screensForRectangle = Screen.getScreensForRectangle(windowCenter);
-    return screensForRectangle.get(0).getVisualBounds();
+    return JavaFxUtil.getScreenForRectangle(
+        new Rectangle2D(stage.getX(),
+            stage.getY(),
+            stage.getWidth(),
+            stage.getHeight())).getVisualBounds();
```

Comment:
_Why_ was this necessary? I can't believe that Java's `Screen.getScreensForRectangle()` doesn't work correctly.

Reply:
I thruw nullpointer Exceptions

Reply:
occaisonally